lib/storage/acl.js

```diff
@@ -328,3 +451,90 @@ Acl.prototype.makeReq_ = function(method, path, query, body, callback) {
 };
 
 module.exports = Acl;
+
+/**
+ * Attach functionality to a {module:storage/acl} instance. This will add an
+ * object for each role group (owners, readers, and writers), with each object
+ * containing methods to add or delete a type of entity.
+ *
+ * As an example, here are a few methods that are created.
+ *
+ *   myBucket.acl.readers.deleteGroup('groupId', function(err) {});
+ *
+ *   myBucket.acl.owners.addUser('email@example.com', function(err, acl) {});
+ *
+ *   myBucket.acl.writers.addDomain('example.com', function(err, acl) {});
+ *
+ * @private
+ */
+function AclRoleAccessorMethods() {
+  AclRoleAccessorMethods.roles.forEach(this._assignAccessMethods.bind(this));
+}
+
+AclRoleAccessorMethods.accessMethods = [
+  'add',
+  'delete'
+];
+
+AclRoleAccessorMethods.entities = [
+  // Special entity groups that do not require further specification.
+  'allAuthenticatedUsers',
+  'allUsers',
+
+  // Entity groups that require specification, e.g. `user-email@example.com`.
+  'domain-',
+  'group-',
+  'project-',
+  'user-'
+];
+
+AclRoleAccessorMethods.roles = [
+  'OWNER',
+  'READER',
+  'WRITER'
+];
+
+AclRoleAccessorMethods.prototype._assignAccessMethods = function(role) {
+  var that = this;
+
+  var accessMethods = AclRoleAccessorMethods.accessMethods;
+  var entities = AclRoleAccessorMethods.entities;
+  var roleGroup = role.toLowerCase() + 's';
+
+  this[roleGroup] = entities.reduce(function(acc, entity) {
+    var isPrefix = entity.substr(-1) === '-';
```

Comment:
It's more readable to use `entity.charAt(entity.length-1)` (and it's [faster](http://jsperf.com/splice-1-vs-charat-lenght-1-vs-index)!)